token-swap/program/src/curve/stable.rs

```diff
@@ -412,20 +434,84 @@ mod tests {
 
     #[test]
     fn constant_product_swap_no_fee() {
-        let source_amount: u128 = 100;
-        let swap_source_amount: u128 = 1000;
-        let swap_destination_amount: u128 = 50000;
-        let curve = StableCurve {
-            amp: 1,
-            ..Default::default()
-        };
-
-        let result = curve
-            .swap(source_amount, swap_source_amount, swap_destination_amount)
-            .unwrap();
-        assert_eq!(result.new_source_amount, 1100);
-        assert_eq!(result.amount_swapped, 2083);
-        assert_eq!(result.new_destination_amount, 47917);
+        const POOL_AMOUNTS: &[u128] = &[
+            100,
+            10_000,
+            1_000_000,
+            100_000_000,
+            10_000_000_000,
+            1_000_000_000_000,
+            100_000_000_000_000,
+            10_000_000_000_000_000,
+            1_000_000_000_000_000_000,
+        ];
+        const SWAP_AMOUNTS: &[u128] = &[
+            100,
+            1_000,
+            10_000,
+            100_000,
+            1_000_000,
+            10_000_000,
+            100_000_000,
+            1_000_000_000,
+            10_000_000_000,
+            100_000_000_000,
+        ];
+        const AMP_FACTORS: &[u64] = &[1, 10, 20, 50, 75, 100, 125, 150];
+
+        for swap_source_amount in POOL_AMOUNTS {
+            for swap_destination_amount in POOL_AMOUNTS {
+                for source_amount in SWAP_AMOUNTS {
+                    for amp in AMP_FACTORS {
+                        let curve = StableCurve {
+                            amp: *amp,
+                            ..Default::default()
+                        };
+
+                        if *source_amount >= *swap_source_amount {
+                            continue;
+                        }
+
+                        println!(
+                            "trying: source_amount={}, swap_source_amount={}, swap_destination_amount={}", 
+                            source_amount, swap_source_amount, swap_destination_amount
+                        );
```

Comment:
nit: You've done a great job with the print in the assert, so this can be removed